nilearn/datasets/atlas.py

```diff
@@ -511,11 +725,28 @@ def fetch_atlas_harvard_oxford(
     new_atlas_niimg = new_img_like(
         atlas_niimg, new_atlas_data, atlas_niimg.affine
     )
+
+    if atlas_type == "probabilistic":
+        return Bunch(
```

Comment:
In order to avoid doing the same checks and writing the same code twice, I would suggest to put the necessary check for the below code, above the line `if not symmetric_split or is_lateralized:`
```
    new_atlas_data, new_names = _compute_symmetric_split(
        "HarvardOxford", atlas_niimg, names
    )
    new_atlas_niimg = new_img_like(
        atlas_niimg, new_atlas_data, atlas_niimg.affine
    )
```

and to put `names` and `atlas_niimg` instead of `new_names` and `new_atlas_niimg` variables.